File: 2022/Day10/Puzzel2.py
def crt( cycle, registerValue ):
    linePos = cycle%40
    if( linePos == registerValue-1 or linePos == registerValue or linePos == registerValue+1 ):
        print('#', end='')
    else:
        print('.', end='')
    if( (cycle+1)%40==0 ):
        print('')

input = []
with open("input.txt", "r") as inputFile:
    input = inputFile.readlines()

registerValue = 1
cycle = 0

for line in input:
    line = line.split(" ")
    if( line[0] == "addx" ):
        for i in range(2):
            crt( cycle, registerValue )
            cycle += 1
        registerValue += int(line[1])
    else:
        crt( cycle, registerValue )
        cycle += 1

File: 2022/Day10/Puzzel2.cpp
#include <fstream>
#include <iostream>
#include <string>
#include "StringUtils.hpp"
#include <vector>

void crt( int cycle, const int &registerValue )
{
    int linePos = cycle%40;
    if( linePos == registerValue-1 | linePos == registerValue | linePos == registerValue+1 )
        std::cout << '#';
    else
        std::cout << '.';

    if( (cycle+1)%40==0 )
        std::cout << '\n';
}

int main()
{
    std::ifstream inputFile("input.txt");
    std::string temp;
    int registerValue = 1;
    int cycle = 0;
    while( std::getline( inputFile, temp ) )
    {
        std::vector<std::string> line = StringUtils::splitString( temp, ' ' );
        if( line[0] == "addx" )
        {
            for( unsigned int i = 0; i<2; i++ )
            {
                crt(cycle, registerValue);
                cycle++;
            }
            registerValue += std::stoi( line[1] );
        }
        else
        {
            crt(cycle, registerValue);
            cycle++;
        }    
    }
}